### PR TITLE
Translate Connecting Applications with Services in Bahasa Indonesia

### PR DESCRIPTION
**Pull request by zufardhiyaulhaq**
This PR addresses #13929

### CODE REVIEW

**Comment by zufardhiyaulhaq** [comment]
Sorry, I forget to check first. Let me fix first.

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+content_template: templates/concept
+weight: 30
+---
+
+
+{{% capture overview %}}
+
+## Model Kubernetes untuk menghubungkan kontainer
+
+Sekarang kamu memiliki aplikasi yang telah direplikasi, kamu dapat mengeksposnya di jaringan. Sebelum membahas pendekatan jaringan di Kubernetes, akan lebih baik jika kamu paham bagaimana jaringan bekerja didalam *Docker*.
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: I wonder if we have another wording choice for replicated here hmm

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+
+Secara *default*, *Docker* menggunakan jaringan *host*, jadi kontainer dapat berkomunikasi dengan kontainer lainnya jika mereka berada didalam *node* yang sama. Agar kontainer *Docker* dapat berkomunikasi antar *node*, masing-masing kontainer tersebut harus diberikan *port* yang berbeda di alamat *IP node* tersebut, yang akan diteruskan atau di*proxy*kan kedalam kontainer. Artinya adalah para kontainer didalam sebuah *node* harus berkoordinasi *port* mana yang akan digunakan atau dialokasikan secara otomatis.
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: IP *node*

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+Akan sulit untuk mengkoordinasikan *port* yang digunakan oleh banyak pengembang. Kubernetes mengansumsikan bahwa *Pod* dapat berkomunikasi dengan *Pod* lain, terlepas di *Node* mana *Pod* tersebut di *deploy*. Kubernetes memberikan setiap *Pod* alamat *Cluster IP* sehingga kamu tidak perlu secara explisit membuat jalur antara *Pod* ataupun memetakan *port* kontainer kedalam *port* didalam *Node* tersebut. Ini berarti kontainer didalam sebuah *Pod* dapat berkomunikasi dengan *localhost* via *port*, dan setiap *Pod* didalam kluster dapat berkomunikasi tanpa *NAT*. Panduan ini akan membahas bagaimana kamu dapat menjalankan sebuah layanan atau aplikasi didalam model jaringan diatas.
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: should we use developer or pengembang here?

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+```
+NAME       ENDPOINTS                     AGE
+my-nginx   10.244.2.5:80,10.244.3.4:80   1m
+```
+
+Kamu sekarang dapat melakukan *curl* kedalam *nginx Service* di `<CLUSTER-IP>:<PORT>` dari *node* manapun di kluster. Perlu dicatat bahwa *Service IP* adalah *IP virtual*, *IP* tersebut tidak pernah ada di *interface node* manapun. Jika kamu penasaran bagaimana konsep ini bekerja, kamu dapat membaca lebih lanjut tentang [service proxy](/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies).
+
+## Mengakses Service
+
+Kubernetes mendukung 2 mode utama untuk menemukan sebuah *Service* - variabel *Environment* dan *DNS*.
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: I remember that we agree on using variabel `environment` based on slack discussion

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+
+kubectl get pods -l run=my-nginx -o wide
+```
+```
+NAME                        READY     STATUS    RESTARTS   AGE     IP            NODE
+my-nginx-3800858182-e9ihh   1/1       Running   0          5s      10.244.2.7    kubernetes-minion-ljyd
+my-nginx-3800858182-j4rm4   1/1       Running   0          5s      10.244.3.8    kubernetes-minion-905m
+```
+
+Kamu melihat *Pod* memiliki nama yang berbeda, karena mereka dihapus dan dibuat ulang.
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: Kamu mungkin saja melihat `Pod` dengan nama yang berbeda, hal ini terjadi karena `Pod-Pod` itu dihapus dan dibuat ulang.

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+NAME       TYPE        CLUSTER-IP     EXTERNAL-IP        PORT(S)               AGE
+my-nginx   ClusterIP   10.0.162.149   162.222.184.144    80/TCP,81/TCP,82/TCP  21s
+```
+```
+curl https://<EXTERNAL-IP> -k
+...
+<title>Welcome to nginx!</title>
+```
+
+IP address pada kolom `EXTERNAL-IP` menujukan *IP* yang tersedia di internet. Sedangkan kolom `CLUSTER-IP* merupakan *IP* yang hanya tersedia didalam kluster kamu (*IP private*).
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: ni: alamat IP
Please change all *IP* with IP

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+
+Perhatikan pada *AWS*, tipe `LoadBalancer` membuat sebuah *ELB*, yang menggunakan *hostname* yang panjang, bukan *IP*. Karena tidak semua keluar pada standar keluaran `kubectl get svc`. Jadi kamu harus menggunakan `kubectl describe service my-nginx` untuk melihatnya. Kamu akan melihat seperti ini:
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: Any other suggestions on how we translate `*hostname* yang panjang`

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+...
+LoadBalancer Ingress:   a320587ffd19711e5a37606cf4a74574-1142138393.us-east-1.elb.amazonaws.com
+...
+```
+
+{{% /capture %}}
+
+{{% capture whatsnext %}}
+
+Kubernetes juga mendukung *Federated Services*, yang bisa mempengaruhi banyak kluster dan penyedia layanan *cloud*, untuk meningkatkan ketersediaan, peningkatan toleransi kesalahan, dan pengembangan dari *Service* kamu. Lihat [Panduan Federated Services](/docs/concepts/cluster-administration/federation-service-discovery/) untuk informasi lebih lanjut.
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: `Panduan `Federated Service``

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+
+- Didalam file tersebut, ada spesifikasi *Deployment* dan *Service*
+- ada [server nginx](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/https-nginx/default.conf) yang melayani trafik *HTTP* di *port* 80 dan trafik *HTTPS* di *port* 443, dan *Service nginx* yang mengekspos kedua *port* tersebut.
+- Setiap kontainer mempunyai akses ke *key* melalui *volume* yang di *mount* pada `/etc/nginx/ssl`. Ini adalah konfigurasi sebelum server *nginx* dijalankan.
+
+```shell
+kubectl delete deployments,svc my-nginx; kubectl create -f ./nginx-secure-app.yaml
+```
+
+Pada titik ini, kamu dapat berkomunikasi dengan server *nginx* dari *node* manapun.
```
Comment (irvifa) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: Pada tahapan ini

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 10]
```diff
@@ -0,0 +1,366 @@
...
+Kamu sekarang dapat melakukan *curl* kedalam *nginx Service* di `<CLUSTER-IP>:<PORT>` dari *node* manapun di kluster. Perlu dicatat bahwa *Service IP* adalah *IP virtual*, *IP* tersebut tidak pernah ada di *interface node* manapun. Jika kamu penasaran bagaimana konsep ini bekerja, kamu dapat membaca lebih lanjut tentang [service proxy](/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies).
+
+## Mengakses Service
+
+Kubernetes mendukung 2 mode utama untuk menemukan sebuah *Service* - variabel *Environment* dan *DNS*. 
+*DNS* membutuhkan [tambahan CoreDNS didalam kluster](http://releases.k8s.io/{{< param "githubbranch" >}}/cluster/addons/dns/coredns).
+
+### Variabel Environment
+
+Saat sebuah *Pod* berjalan di *Node*, *kubelet* akan menambahkan variabel *Environment* untuk setiap *Service* yang aktif kedalam *Pod*. Ini menimbulkan beberapa masalah. Untuk melihatnya, periksa *environment* dari *Pod nginx* yang telah kamu buat (nama *Pod*mu akan berbeda-beda):
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: *Pod*mu doesn't work out for markdown. Probably: *Pod*-mu

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 11]
```diff
@@ -0,0 +1,366 @@
...
+kubectl get services kube-dns --namespace=kube-system
+```
+```
+NAME       TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)         AGE
+kube-dns   ClusterIP   10.0.0.10    <none>        53/UDP,53/TCP   8m
+```
+
+Jika *DNS* belum berjalan, kamu dapat [mengaktifkannya](http://releases.k8s.io/{{< param "githubbranch" >}}/cluster/addons/dns/kube-dns/README.md#how-do-i-configure-it).
+
+Sisa panduan ini mengansumsikan kamu mempunyai *Service* dengan *IP* (my-nginx), dan sebuah server *DNS*  yang memberikan nama kedalam *IP* tersebut (CoreDNS kluster),jadi kamu dapat berkomunikasi dengan *Service* dari *Pod* lain didalam kluster menggunakan metode standar (contohnya *gethostbyname*). Jalankan aplikasi *curl* lain untuk melakukan pengujian ini:
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: typo: mengansumsikan -> mengasumsikan

Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: space after comma

..., jadi kamu dapat ...

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 12]
```diff
@@ -0,0 +1,366 @@
...
+
+Name:      my-nginx
+Address 1: 10.0.162.149
+```
+
+## Mengamankan Service
+
+HIngga sekarang kita hanya mengakses *nginx* server dari dalam kluster. Sebelum mengekspos *Service* ke internet, kamu harus memastikan bahwa kanal komunikasi aman. Untuk melakukan hal tersebut, kamu membutuhkan:
+
+* Self signed certificates* untuk *https* (kecuali jika kamu sudah mempunyai *identity certificate*)
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: missing asterisk, should be *Self signed certificates*

File: content/id/docs/concepts/services-networking/connect-applications-service.md (hunk 7 above)
Comment (zufardhiyaulhaq) [reply content/id/docs/concepts/services-networking/connect-applications-service.md]: nama objek yang panjang?

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 13]
```diff
@@ -0,0 +1,366 @@
...
+* Sebuah server *nginx* yang terkonfigurasi untuk menggunakan sertifikat tersebut
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: I think *certificate* is more consistent than sertifikat (with the translated pages) and `self-signed certificate`

same with the rest

File: content/id/docs/concepts/services-networking/connect-applications-service.md (hunk 4 above)
Comment (zufardhiyaulhaq) [reply content/id/docs/concepts/services-networking/connect-applications-service.md]: so `environment` without capital `E`?

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 14]
```diff
@@ -0,0 +1,366 @@
...
+```
+NAME                  TYPE                                  DATA      AGE
+default-token-il9rc   kubernetes.io/service-account-token   1         1d
+nginxsecret           Opaque                                2         1m
+```
+
+Berikut ini adalah langkah-langkah manual yang harus diikuti jika kamu mengalami masalah menjalankan *make* (pada windows contohnya): 
+
+```shell
+#membuat sebuah pasangan kunci public private
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: key-pair is more understandable

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 15]
```diff
@@ -0,0 +1,366 @@
...
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /d/tmp/nginx.key -out /d/tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"
+#rubah kunci tersebut kedalam pengkodean base64
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: kunci -> key, more understandable

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 16]
```diff
@@ -0,0 +1,366 @@
...
+cat /d/tmp/nginx.crt | base64
+cat /d/tmp/nginx.key | base64
+```
+Gunakan hasil keluaran dari perintah sebelumnya untuk membuat sebuah file *yaml* seperti berikut. nilai yang dikodekan *base64* harus berada didalam satu baris.
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: nilai -> Nilai

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 17]
```diff
@@ -0,0 +1,366 @@
...
+
+```shell
+kubectl get pods -o yaml | grep -i podip
+    podIP: 10.244.3.5
+node $ curl -k https://10.244.3.5
+...
+<h1>Welcome to nginx!</h1>
+```
+
+Perlu dicatat bahwa kita menggunakan parameter `-k` saat menggunakan *curl*, ini karena kita tidak tau apapun tentang *Pod* yang menjalankan *nginx* saat pembuatan seritifikat, jadi kita harus memberitahu *curl* untuk mengabaikan ketidakcocokan *CName*. Dengan membuat *Service*, kita menghubungkan *CName* yang digunakan pada sertifikat dengan nama pada *DNS* yang digunakan *Pod*. Lakukan pengujian dari sebuah *Pod* (*secret* yang sama digunakan untuk agar mudah, *Pod* tersebut hanya membutuhkan *nginx.crt* untuk mengakses *Service*)
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: > (*secret* yang sama digunakan untuk agar mudah, *Pod* tersebut hanya membutuhkan *nginx.crt* untuk mengakses *Service*)

this sentence doesn't make sense, probably missing translation?

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 18]
```diff
@@ -0,0 +1,366 @@
...
+```shell
+kubectl exec curl-deployment-1515033274-1410r -- curl https://my-nginx --cacert /etc/nginx/ssl/nginx.crt
+...
+<title>Welcome to nginx!</title>
+...
+```
+
+## Mengekspos Service
+
+Kamu mungkin ingin mengekspos *Service* ke alamat *IP* eksternal. Kubernetes mendukung dua cara untuk melakukan ini: *NodePorts* dan *LoadBalancers*. *Service* yang dibuat tadi sudah menggunakan `NodePort`, jadi replika *nginx* sudah siap untuk menerima trafik dari internet jika *Node* kamu mempunyai *IP* publik.
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: NodePorts -> NodePort; LoadBalancers -> LoadBalancer

(prefer singular over plural)

File: content/id/docs/concepts/services-networking/connect-applications-service.md [hunk 19]
```diff
@@ -0,0 +1,366 @@
...
+      type: ExternalIP
+    allocatable:
+...
+
+$ curl https://<EXTERNAL-IP>:<NODE-PORT> -k
+...
+<h1>Welcome to nginx!</h1>
+```
+
+Mari coba membuat ulang *Service* menggunakan *cloud load balancer*, ubah saja `type`*Service* `my-nginx`  dari `NodePort` ke `LoadBalancer`:
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: nit: missing space

`type`*Service*

should be:

`type` *Service*

File: content/id/docs/concepts/services-networking/connect-applications-service.md (hunk 6 above)
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: typo: menujukan

Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/connect-applications-service.md]: missing backtick:

`CLUSTER-IP* -> `CLUSTER-IP`

File: content/id/docs/concepts/services-networking/connect-applications-service.md (hunk 17 above)
Comment (zufardhiyaulhaq) [reply content/id/docs/concepts/services-networking/connect-applications-service.md]: I am translating from `the same secret is being reused for simplicity, the pod only needs nginx.crt to access the Service`.

Comment (girikuncoro) [reply content/id/docs/concepts/services-networking/connect-applications-service.md]: let's translate this into:
```
*secret* yang sama digunakan kembali untuk penyederhanaan, ...
```

**Comment by girikuncoro** [comment]
we can do follow up PR later for the rest (as it's on going discussion and style still being standardized), I'm going to merge this for now

/lgtm
/approve

**Pull request merged**